p2p/peer.go

```diff
@@ -119,15 +119,15 @@ func newOutboundPeerConn(addr *NetAddress, config *PeerConfig, persistent bool,
 
 	pc, err = newPeerConn(conn, config, true, persistent, ourNodePrivKey)
 	if err != nil {
-		if err2 := conn.Close(); err != nil {
+		if err2 := conn.Close(); err2 != nil {
```

Comment:
:o !!! thanks

Reply:
is there a good paradigm/tactic we can use to avoid this kind of thing ?

Reply:
always use `err`? `metalinter`?